### PR TITLE
Enable CRCs in the protocol

### PR DESCRIPTION
**Pull request by pnarayanan**
This is a follow-up of #549 which introduced new versions with CRCs
and the capability to read them correctly. This patch enables these
new versions.

Fixes linkedin/ambry#454

### CODE REVIEW

**Comment by pnarayanan** [comment]
@nsivabalan yes

File: ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java [hunk 1]
```diff
@@ -99,9 +98,7 @@ public static ReceivedPutRequest readFrom(DataInputStream stream, ClusterMap map
   @Override
   public long sizeInBytes() {
     long sizeInBytes = sizeExcludingBlobAndCrcSize() + blobSize;
-    if (currentVersion == Put_Request_Version_V3) {
```
Comment (vgkholla) [inline ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java]: there is a similar check to remove in the `writeTo()` method

File: ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java [hunk 2]
```diff
@@ -57,8 +57,7 @@
   private static final short Put_Request_Version_V2 = 2;
```
Comment (vgkholla) [inline ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java]: you could remove this declaration if the frontends are first updated and then the servers? or are you planning to do cleanup in one shot?

Comment (pnarayanan) [inline ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java]: yeah, I think cleanup can be done in a subsequent PR.

File: ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java (hunk 1 above)
Comment (pnarayanan) [inline ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java]: removed.

**Comment by pnarayanan** [comment]
@vgkholla updated, built and tested.

**Pull request merged**